pulser-core/pulser/noise_model.py

```diff
@@ -274,6 +299,65 @@ def to_tuple(obj: tuple) -> tuple:
                     stacklevel=2,
                 )
 
+    def _register_sigma_xy_z(self) -> tuple[float, float]:
+        """Standard deviation for fluctuations in atom position in the trap.
+
+        - Plane fluctuation: 𝜎ˣʸ = √(T w²/(4 Uₜᵣₐₚ)), where T is temperature,
+          w is the trap waist and Uₜᵣₐₚ is the trap depth.
+        - Off plane fluctuation: 𝜎ᶻ = 𝜋 / 𝜆 √2 w 𝜎ˣʸ, where 𝜆 is the trap
+        wavelength with a constant value of 0.85 µm
+
+        Note: a k_B factor is absorbed in the trap depth (Uₜᵣₐₚ), so the units
+        of temperature and trap depth are the same.
+
+        Args:
+            temperature (float): Temperature (T) of the atoms in the trap
+            (in Kelvin).
+            trap_depth (float): Depth of the trap (Uₜᵣₐₚ)
+            (same units as temperature).
+            trap_waist (float): Waist of the trap (w) (in µmeters).
+
+        Returns:
+            tuple: The standard deviations of the spatial position fluctuations
+            in the xy-plane (register_sigma_xy) and along the z-axis
+            (register_sigma_z).
+        """
+        tr_depth = cast(float, self.trap_depth)  # trap_depth is not None here
+        register_sigma_xy = math.sqrt(
+            self.temperature * self.trap_waist**2 / (4 * tr_depth)
+        )
+        register_sigma_z = (
+            math.pi
+            / TRAP_WAVELENGTH
+            * math.sqrt(2)
+            * self.trap_waist
+            * register_sigma_xy
+        )
+        return register_sigma_xy, register_sigma_z
+
+    @staticmethod
+    def _check_register_noise_params(
+        true_noise_types: Collection[NoiseTypes],
+        trap_waist: float,
+        trap_depth: float | None,
+        temperature: float,
+    ) -> None:
+        if "register" not in true_noise_types:
+            return
+        if trap_waist == 0.0 or trap_depth is None:
+            raise ValueError(
+                "trap_waist and trap_depth must be defined in "
+                "order to simulate register noise."
+            )
+        elif temperature == 0.0:
+            warnings.warn(
+                "trap_waist, trap_depth, and temperature must be defined in "
+                "order to simulate register noise. Temperature is "
+                f"{temperature} µK. Register noise will have no effect",
+                UserWarning,
+                stacklevel=2,
+            )
```

Comment:
@a-quelle-pasqal @murogrande I know we agreed on this and it will work, but it can be quite inefficient as it will trigger the "noisy simulation" even when `temperature==0.0`(so there is actually no noise). Given that `"register"` noise is activated iff `"trap_waist"` or `"trap_depth"` are given, can't we just enforce that all three parameters must be given in that case?

```suggestion
        if trap_waist == 0.0 or trap_depth is None or temperature == 0.0:
            raise ValueError(
                "trap_waist, trap_depth and temperature must all"
                " be defined in order to simulate register noise."
            )
```